internal/service/url.go

```diff
@@ -0,0 +1,64 @@
+package service
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/GlebRadaev/shlink/internal/repository"
+	"github.com/GlebRadaev/shlink/internal/utils"
+)
+
+const (
+	MaxIDLength  = 8
+	MaxURLLength = 2048
+)
+
+// URLService handles the business logic for shortening URLs
+type URLService struct {
+	storage *repository.MemoryStorage
+}
+
+// NewURLService creates a new URLService
+func NewURLService(storage *repository.MemoryStorage) *URLService {
+	return &URLService{storage: storage}
+}
+
+// ShortenURL shortens a given URL and returns the short version
+func (s *URLService) Shorten(url string) (string, error) {
+	if len(url) > MaxURLLength {
```

Comment:
[SUGGESTION] You can simplify it.
https://stackoverflow.com/questions/31480710/validate-url-with-standard-package-in-go